lib/mpl_toolkits/axisartist/floating_axes.py

```diff
@@ -308,30 +308,15 @@ def get_boundary(self):
 class FloatingAxesBase:
 
     def __init__(self, *args, **kwargs):
-        grid_helper = kwargs.get("grid_helper", None)
-        if grid_helper is None:
-            raise ValueError("FloatingAxes requires grid_helper argument")
-        if not hasattr(grid_helper, "get_boundary"):
-            raise ValueError("grid_helper must implement get_boundary method")
-
         self._axes_class_floating.__init__(self, *args, **kwargs)
-
         self.set_aspect(1.)
         self.adjust_axes_lim()
 
     def _gen_axes_patch(self):
-        """
-        Returns the patch used to draw the background of the axes.  It
-        is also used as the clipping path for any data elements on the
-        axes.
-
-        In the standard axes, this is a rectangle, but in other
-        projections it may not be.
-
-        .. note::
-            Intended to be overridden by new projection types.
-        """
+        # docstring inherited
         grid_helper = self.get_grid_helper()
+        if not hasattr(grid_helper, "get_boundary"):
+            raise ValueError("grid_helper must implement get_boundary method")
```

Comment:
Is it really helpful to move this into a private method? Seems clearer for a user if the exception comes from the class init directly.

Reply:
_gen_axes_patch is called immediately at init so users would see the message immediately, just with a few more stack frames.  Putting the check e.g. after the super()init call would actually be useless because super()init would already have called _gen_axes_patch and thus already errored out.

Reply:
I don't understand. The first thing the original code did in `__init__` was validating the `grid_helper` parameter.